library/src/blas3/rocblas_trsm.hpp

```diff
@@ -0,0 +1,1737 @@
+                                                                 m,
+                                                                 stride_X,
+                                                                 batch_count);
+                        if(i - BLOCK < 0)
+                            break;
+                        rocblas_gemm_template<BATCHED, !BATCHED>(handle,
+                                                                 transB,
+                                                                 transA,
+                                                                 m,
+                                                                 i,
+                                                                 BLOCK,
+                                                                 &negative_one<T>,
+                                                                 0,
+                                                                 (U)X,
+                                                                 i * m,
+                                                                 m,
+                                                                 stride_X,
+                                                                 A,
+                                                                 i * lda + offset_Ain,
+                                                                 lda,
+                                                                 stride_A,
+                                                                 &one<T>,
+                                                                 0,
+                                                                 B,
+                                                                 offset_Bin,
+                                                                 ldb,
+                                                                 stride_B,
+                                                                 batch_count);
+                    }
+                }
+            }
+        } // tranpsose
+
+        return rocblas_status_success;
+    }
+
+    template <rocblas_int BLOCK, bool BATCHED, typename T, typename U, typename V>
+    rocblas_status special_trsm_template(rocblas_handle    handle,
+                                         rocblas_side      side,
+                                         rocblas_fill      uplo,
+                                         rocblas_operation transA,
+                                         rocblas_diagonal  diag,
+                                         rocblas_int       m,
+                                         rocblas_int       n,
+                                         const T*          alpha,
+                                         rocblas_stride    stride_alpha,
+                                         U                 A,
+                                         rocblas_int       offset_Ain,
+       
```

Comment:
We should be avoiding `hipMemcpy`. 

We should be computing it on device rather than host.

Reply:
This introduces synchronization at the end of every `hipMemcpy()` call, and may negate the advantages of batched.

Reply:
This one was introduced by me. I will work on a way around this.